### PR TITLE
Support --epochs instead of --num-steps in optical flow references

### PR DESCRIPTION
**Pull request by NicolasHug**
Towards https://github.com/pytorch/vision/issues/5056

This PR removes support for `--num-steps ` and adds support for `--epochs` instead in the optical flow training reference.

I'm hoping this can get approved and in the mean time I'll re-run some training runs to make sure we still get good results. I will post the final training script once done.

cc @datumbox

### CODE REVIEW

File: references/optical_flow/train.py
Comment (NicolasHug) [inline references/optical_flow/train.py:240]: We had a +100 before.
Without this +1, I'm consistently seeing some weird errors when training on Kitti for 2 epochs:

<details>

```
Traceback (most recent call last):
  File "/fsx/users/nicolashug/work/vision/references/optical_flow/train.py", line 345, in <module>
    main(args)
  File "/fsx/users/nicolashug/work/vision/references/optical_flow/train.py", line 254, in main
    train_one_epoch(
  File "/fsx/users/nicolashug/work/vision/references/optical_flow/train.py", line 173, in train_one_epoch
    for data_blob in logger.log_every(train_loader):
  File "/fsx/users/nicolashug/work/vision/references/optical_flow/utils.py", line 123, in log_every
    for obj in iterable:
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 521, in __next__
    data = self._next_data()
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1203, in _next_data
    return self._process_data(data)
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 1229, in _process_data
    data.reraise()
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/_utils.py", line 438, in reraise
    raise exception
RuntimeError: Caught RuntimeError in DataLoader worker process 9.
Original Traceback (most recent call last):
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/_utils/worker.py", line 287, in _worker_loop
    data = fetcher.fetch(index)
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    return self.collate_fn(data)
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 100, in default_collate
    return [default_collate(samples) for samples in transposed]  # Backwards compatibility.
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 100, in <listcomp>
    return [default_collate(samples) for samples in transposed]  # Backwards compatibility.
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/utils/data/_utils/collate.py", line 64, in default_collate
    storage = elem.storage()._new_shared(numel)
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/storage.py", line 486, in _new_shared
    untyped_storage = module.UntypedStorage._new_shared(size * cls().element_size())
  File "/fsx/users/nicolashug/conda/envs/raft/lib/python3.9/site-packages/torch/storage.py", line 173, in _new_shared
    return cls._new_using_fd(size)
RuntimeError: unable to open shared memory object </torch_24843_0> in read-write mode: File exists (17)
```

</details>

File: train.sh
Comment (NicolasHug) [inline train.sh:1]: Pushing for my own convenience during the training, but I'll remove before merging

File: references/optical_flow/train.py
Comment (datumbox) [inline references/optical_flow/train.py:305]: Why the change of the default value of batch-size from 6 to 2?

File: references/optical_flow/train.py
Comment (datumbox) [inline references/optical_flow/train.py:304]: Shouldn't the default epochs be 20 given your estimations on the readme?

File: references/optical_flow/train.py
Comment (datumbox) [reply references/optical_flow/train.py:240]: This is really concerning but I don't know what could cause this. @fmassa do you have any thoughts on what this could be?

**Review by datumbox — COMMENTED** [review]
Thanks @NicolasHug, overall the approach looks correct. I've added some questions, I would love to understand better some of the changes.

File: references/optical_flow/train.py
Comment (NicolasHug) [reply references/optical_flow/train.py:305]: It's unrelated to the PR but all the pretrained weights have relied on bs=2 so I figured I should update it

File: references/optical_flow/train.py
Comment (NicolasHug) [reply references/optical_flow/train.py:304]: I'll change to 20, no strong opinion. The value that is used varies from 7 to 4000 in the training script.

**Review by datumbox — APPROVED** [review]
Approving to unblock. Overall, I think the changes look OK. Issues related to resuming the scheduler can be fixed by serializing the scheduler and optimizer in the checkpoint as @fmassa indicated on a previous PR. This can be addressed on a followup PR.

Concerning the weird error discussed at https://github.com/pytorch/vision/pull/5082/files#r765914829, this might be worth investigating more as the error message looks weird.

File: references/optical_flow/train.py
Comment (NicolasHug) [reply references/optical_flow/train.py:240]: I'm unable to reproduce the error today. I'm suspecting that this is related to the state of my env at the time. I do observe some rare ddp failures from time to time, so I'm hoping that this was just one of these one-off events.

I ran the training with `epochs + 1` and got good epe resuts on C+T. I will remove the `+ 1` and train again. If I still get good resuts, and if I still can't reproduce, I think it'll be safe to merge.

**Pull request merged**
